File: sample/python/turtle_spiral.py
from math import pi
from build_box import BuildBox
from turtle import Turtle

room_name = "1000"
build_box = BuildBox(room_name)

build_box.set_box_size(0.3)
build_box.set_build_interval(0.001)
build_box.set_command('liteRender')
t = Turtle(build_box)

colors = [
  [0, 0, 0, 1],
  [1, 0, 0, 1],
  [0, 1, 0, 1],
  [0, 0, 1, 1],
  [1, 1, 0, 1],
  [0, 1, 1, 1],
  [1, 0, 1, 1],
#   [1, 1, 1, 1],
#   [0.5, 0, 0, 1],
#   [0, 0.5, 0, 1],
#   [0, 0, 0.5, 1],
#   [0.5, 0.5, 0, 1],
#   [0, 0.5, 0.5, 1],
#   [0.5, 0, 0.5, 1],
#   [0.5, 0.5, 0.5, 1],
]

for i, color in enumerate(colors):
  t.reset()
  t.set_color(*color)
  t.set_pos(i, 0, 0)
  t.up(4)

  for _ in range(360):
    t.forward(3)
    t.left(6)

build_box.send_data()

File: sample/javaScript/turtleSpiral.mjs
import { Voxelamming, Turtle } from 'voxelamming';
// import { Turtle } from 'voxelamming'; // test
// import Voxelamming from './voxelamming.js';  // test

const roomName = "1000";
const vox = new Voxelamming(roomName);

vox.setBoxSize(0.3);
vox.setBuildInterval(0.001);
vox.setCommand('liteRender')
const t = new Turtle(vox);

const colors = [
  [0, 0, 0, 1],
  [1, 0, 0, 1],
  [0, 1, 0, 1],
  [0, 0, 1, 1],
  [1, 1, 0, 1],
  [0, 1, 1, 1],
  [1, 0, 1, 1],
  // [1, 1, 1, 1],
  // [0.5, 0, 0, 1],
  // [0, 0.5, 0, 1],
  // [0, 0, 0.5, 1],
  // [0.5, 0.5, 0, 1],
  // [0, 0.5, 0.5, 1],
  // [0.5, 0, 0.5, 1],
  // [0.5, 0.5, 0.5, 1],
];

for (let i = 0; i < colors.length; i++) {
  const color = colors[i];
  const polarPhi = (i * 180) / colors.length;
  t.reset();
  t.setColor(...color);
  t.setPos(i, 0, 0);
  t.up(4);

  for (let _ = 0; _ < 360; _++) {
    t.forward(3);
    t.left(6);
  }
}

await vox.sendData();
console.log('send data done');
